apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-template-modal/components/template-selector-item.js

```diff
@@ -31,6 +30,13 @@ const TemplateSelectorItem = ( props ) => {
 		return null;
 	}
 
+	const staticPreviewImg =
+		'blank' === value
+			? null
+			: 'https://s0.wordpress.com/mshots/v1/' +
```

Comment:
With using mshots directly, how do we handle the case of the image being generated? Currently mshots displays the loading gif, and in other uses of mshots the user must reload the page again before they see the rendered screenshot. I had a go at doing this in #43428, but the approach I tried out is a bit unreliable, but there could be something useful in there?

In terms of how mshots caches things, it sounds like most of the time users should see the cached image (p3topS-Pr-p2#comment-3379), but in practice I can assume that some users some of the time will see the loader, and it'd be good to make sure it doesn't look like the page is stuck.

Reply:
Thanks @andrewserong 

I'll also add that the demo endpoint works for the demo endpoint without any query parameters, e.g., https://s0.wordpress.com/mshots/v1/https://public-api.wordpress.com/rest/v1/template/demo/hever/dotcompatterns.wordpress.com?vpw=1024&vph=1024&w=500&h=500

But `?language=de` and other accepted params I think are stripped out in class-mshots.php (in gh-Automattic/mShots). 

I mean it does something like this:

```
$snapshot_url = substr(parse_url('https://s0.wordpress.com/mshots/v1/https://public-api.wordpress.com/rest/v1/template/demo/hever/dotcompatterns.wordpress.com?post_id=1340&language=en/?vpw=1024&vph=1024&w=500&h=500')[ 'path' ], 11);
> "https://public-api.wordpress.com/rest/v1/template/demo/hever/dotcompatterns.wordpress.com"
```
So unless we tweak either mshots (to accept the snapshot url query params) or the demo endpoint (to use route fragments???), we won't have translated mshots.

It'd be a very nice thing to have!

Reply:
I've added an `onLoad` event to each thumbnail that reloads the image after 3 seconds. This is more than enough time to grab the final created image since it's so small.

Reply:
The params seem to be okay with mshots since the post_id is working fine.

Reply:
> The params seem to be okay with mshots since the post_id is working fine.

Oh nice! I think I forgot to take into account `encodeURIComponent` 🎉 

Looks like the mshots are being translated where the context exists

<img width="591" alt="Screen Shot 2020-12-22 at 3 50 32 pm" src="https://user-images.githubusercontent.com/6458278/102850428-9103c700-446d-11eb-8ebb-c06e6f03d848.png">